### ISSUE TITLE
"Access to the path is denied" error when launching Bravo

### ISSUE THREAD

**Issue by rudzimj** [sql-bi/Bravo#459]
### Describe the problem

When I start Bravo from Windows or from External Tools, it immediately dies with this error:

```
System.UnauthorizedAccessException: Access to the path is denied.
   at System.IO.Pipes.NamedPipeClientStream.TryConnect(Int32 timeout, CancellationToken cancellationToken)
   at System.IO.Pipes.NamedPipeClientStream.ConnectInternal(Int32 timeout, CancellationToken cancellationToken, Int32 startTime)
   at System.IO.Pipes.NamedPipeClientStream.Connect(Int32 timeout)
   at Sqlbi.Bravo.Infrastructure.AppInstance.NotifyOwner() in D:\a\1\s\src\Infrastructure\AppInstance.cs:line 83
   at Sqlbi.Bravo.Program.Main() in D:\a\1\s\src\Program.cs:line 33
```

It is likely caused by security software. What is Bravo trying to connect to at startup?

### Bravo version

Bravo.0.9.5.x64.msi and Bravo.0.9.5.x64-userinstaller.msi

### Tabular model

Power BI Desktop, import mode

### Power BI

Power BI Desktop Version: 2.106.582.0 64-bit (June 2022)

### Steps to reproduce the behavior

1. Installed Bravo
2. Launched Bravo from Start Menu  or  Power BI Desktop > External Tools
3. Immediately receive error

### Additional context

We use a lot of security apps such Symantec and DataLocker PortBlocker.

### Screenshots

<img width="386" alt="image" src="https://user-images.githubusercontent.com/36211641/181312576-bfc4e79e-d44d-491b-a510-4fa6cfbed067.png">

**Comment by albertospelta** [comment]
@rudzimj Bravo allows only one instance of the application to run at a time and for this reason Bravo checks at startup if another instance is already running. This is based only on communication between local applications/processes and no external or network connections are made.

Unfortunately, we're not able to reproduce this issue. Could you check in the Windows Event Viewer and look for any error around the time of this error in the `Application Logs` ? Please share the event details here. Thanks!

**Comment by albertospelta** [comment]
@rudzimj we have an internal build available that tries to fix the error, it would be great if you could test it and let us know if the issue still persists. 

The build ID is `20220821.1-0.9.8268.26447` and you can download it at this address: https://bravo.bi/builds 

- portable version https://bravorelease.blob.core.windows.net/bravo-internal/2022/08/21/20220821.1-0.9.8268.26447/Bravo.0.9.5.x64-portable.zip
- user installer version https://bravorelease.blob.core.windows.net/bravo-internal/2022/08/21/20220821.1-0.9.8268.26447/Bravo.0.9.5.x64-userinstaller.msi

**Comment by rudzimj** [comment]
I tried launching Bravo before and after this build and I get the same (but different than before) message:
![image](https://user-images.githubusercontent.com/36211641/186166171-7be9d4f9-0b83-4ce4-ba0f-563f20396d02.png)

If I choose **Automatic** I get this message:
![image](https://user-images.githubusercontent.com/36211641/186166283-bc66dbac-5eb1-4b4f-9669-89ca0751dd60.png)

It looks like Bravo tries to start Edge with Admin rights. That won't work here. Here's the event in the Application Log:
<img width="583" alt="image" src="https://user-images.githubusercontent.com/36211641/186167098-1caac46b-a32a-40bf-8f11-121aac7d12be.png">

**Comment by rudzimj** [comment]
I'm seeing messages in Power BI and now the PowerPoint Add-in that indicate WebView2 was not successfully installed. I'm trying to resolve this but not having a lot of luck at the moment. This is what I get when I run the WebView2 installer:
<img width="128" alt="image" src="https://user-images.githubusercontent.com/36211641/186705362-d1008cb1-4dce-4e2b-8d13-4191ee9f311f.png">

**Comment by rudzimj** [comment]
I had to uninstall WebView2 and then reinstall without elevation as an admin, and now Bravo works. 

It looks like the install failure left WebView2 in an odd state. If I ran the installer without admin rights, it thought WebView2 was installed. If I ran the installer with admin rights, it would fail with no message on screen, just the 0x80004005 error in the Event Log. The uninstall cleaned it up, but the install as admin still failed the same way. After an another uninstall, installing without admin rights was slower, but worked perfectly.

Let me know if you need anything else to document this error, but you can close this out.

**Comment by albertospelta** [comment]
@rudzimj when you run the installer as admin you are asked to enter the username and password or just to confirm (yes/no button)?

**Comment by rudzimj** [comment]
I get a prompt from **BeyondTrust Privilege Management** software that I select a reason for the install and click OK.

**Comment by albertospelta** [comment]
Thank you so much for the detailed information you have provided.

I would like to do some further test because I think the initial error - the UnauthorizedAccessException - still exists.

Which version of bravo are you using now?  The initial version downloaded from GitHub or the internal test build?  Which MSI installer file did you use (userinstaller / portable / other)?

Can you try to run Bravo with admin rights and then run a second instance without admin rights?

**Comment by rudzimj** [comment]
Here's the original install:
<img width="283" alt="image" src="https://user-images.githubusercontent.com/36211641/186751234-ea0d43f6-8152-4ca9-9624-791810f30426.png">

And here's the build you requested me to install:
<img width="279" alt="image" src="https://user-images.githubusercontent.com/36211641/186751311-cf686111-036e-43b8-872e-4230173a2c01.png">

This is what I see in my Downloads folder:
<img width="391" alt="image" src="https://user-images.githubusercontent.com/36211641/186751569-e20baffe-29d7-4591-97d1-0131a65ead02.png">
It looks like I tried the msi and then the userinstaller.

To be clear, when I was talking about admin rights today, I was just talking about WebView2 install issues. I did not do any reinstalls of Bravo of today. I'm guessing the first error was related to the WebView2 install. I don't have to run Bravo now with admin rights for it to work. Both builds work now that WebView2 is installed properly.

I'm happy to de-install these 2 versions and run whatever install process or version you want.

**Comment by albertospelta** [comment]
> I'm happy to de-install these 2 versions and run whatever install process or version you want.
Yes, it would be great if you could de-install those 2 versions test this new internal build.
Thanks!

https://bravorelease.blob.core.windows.net/bravo-internal/2022/08/25/20220825.1-0.9.8272.26313/Bravo.0.9.5.x64.msi

**Comment by rudzimj** [comment]
I de-installed both versions and installed this build and it launched clean. 

Then I de-installed Bravo **and** WebView2 and ran again. Bravo installed WebView2 and it worked fine.

Then I realized that I did it without running the installer as admin, so I de-installed both and tried again. There is no option to run as admin on the right click menu, but there was an **Install** option, so I used that. It had a prompt that this would require elevation. I'm not sure if the previous ones did... I didn't pay attention to that. Again Bravo installed WebView2 and everything worked.

I think you have it all sorted!
